dags/utils/gcp.py

```diff
@@ -528,3 +531,152 @@ def normalize_table_id(table_name):
+    :param List[str] command:      [Required] Command to run
+    :param str docker_image:       [Required] docker image to use
+    :param str aws_conn_id:        Airflow connection id for AWS access
+    :param str gcp_conn_id:        Airflow connection id for GCP access
+    :param str gke_location:       GKE cluster location
+    :param str gke_cluster_name:   GKE cluster name
+    :param str gke_namespace:      GKE cluster namespace
+    :param str image_pull_policy:  Kubernetes policy for when to pull
+                                   docker_image
+    :param Dict[str, Any] kwargs:  Additional keyword arguments for
+                                   GKEPodOperator
+
+    :return: GKEPodOperator
+    """
+    kwargs["name"] = kwargs.get("name", task_id.replace("_", "-"))
+    return GKEPodOperator(
+        task_id=task_id,
+        gcp_conn_id=gcp_conn_id,
+        project_id=GoogleCloudBaseHook(gcp_conn_id=gcp_conn_id).project_id,
+        location=gke_location,
+        cluster_name=gke_cluster_name,
+        namespace=gke_namespace,
+        image=docker_image,
+        arguments=command,
+        env_vars={
+            key: value
+            for key, value in zip(
+                ("AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"),
+                AwsHook(aws_conn_id).get_credentials() if aws_conn_id else (),
+            )
+            if value is not None
+        },
+        image_pull_policy=image_pull_policy,
+        **kwargs
+    )
+
+
+def dataproc_spark_job(
+    *,
+    dag_name,
+    cluster_name=None
+    parent_dag_name=None,
+    default_args=None,
+    arguments=None,
+    main_class=None,
+    main_jar=None,
+    metadata=None,
+    gcp_conn_id="google_cloud_derived_datasets",
+    dataproc_spark_jars=None,
+    dataproc_storage_bucket="moz-fx-data-derived-datasets-parquet",
+    dataproc_zone="us-central1-a",
+    num_preemptible_workers=0,
+):
+
+    """ Define a subdag that runs a single spark job.
+
+    :return: airflow.models.DAG
+    """
+    if cluster_name is None:
+        cluster_name = dag_name.replace("_", "-") + "-{{ ds_nodash }}"
+
+    with dataproc_cluster(
+        cluster_name=cluster_name,
+        dag_name=dag_name,
+        default_args=default_args,
+        gcp_conn_id=gcp_conn_id,
+        parent_dag_name=parent_dag_name,
+        storage_bucket=dataproc_storage_bucket,
+        zone=dataproc_zone,
+        metadata=metadata,
+    ) as dag, create_dataproc_cluster, delete_dataproc_cluster:
+        run_dataproc_spark = DataProcSparkOperator(
+            task_id="run_dataproc_spark",
+            main_class=main_class,
+            main_jar=main_jar,
+            dataproc_spark_jars=dataproc_spark_jars,
+            arguments=arguments,
+            cluster_name=cluster_name,
+            gcp_conn_id=gcp_conn_id,
+        )
+
+        create_dataproc_cluster >> run_dataproc_spark >> delete_dataproc_cluster
+
+        return dag
+
+@contextmanager
+def dataproc_cluster(
```

Comment:
Similar creation/deletion/run logic already exists in dags/utils/dataproc.py (https://github.com/mozilla/telemetry-airflow/pull/546), as well as dags/prio/dataproc.py (https://github.com/mozilla/telemetry-airflow/pull/561). Since we were working on these in parallel they haven't been consolidated.

Its a PITA to retest everything after a refactor, but this PR will add a 3rd dataproc wrapper helper.

Reply:
thanks for noticing, i'll see about reusing it from one of those